examples/scale/backend/backend.go

```diff
@@ -40,7 +49,149 @@ func Run() {
 
 	logging.ConfigureLogging(cfg)
 
-	// TODO: This is a placeholder - add the actual implementation.
+	beConn, err := rpc.GRPCClientFromConfig(cfg, "api.backend")
+	if err != nil {
+		logger.Fatalf("failed to connect to Open Match Backend, got %v", err)
+	}
+
+	defer beConn.Close()
+	be := pb.NewBackendClient(beConn)
+
+	feConn, err := rpc.GRPCClientFromConfig(cfg, "api.frontend")
+	if err != nil {
+		logger.Fatalf("failed to connect to Open Match Frontend, got %v", err)
+	}
+
+	defer feConn.Close()
+	fe := pb.NewFrontendClient(feConn)
+
+	var totalMatches int
+	var totalAssigned int
+	var failedAssignment int
+	var totalDeleted int
+	var failedDeletion int
+	startTime := time.Now()
 	mprofiles := profiles.Generate(cfg)
-	logger.Infof("Generated Profiles: %v", mprofiles)
+
+	// Continuously fetch matches, assign matches and delete assigned Tickets.
+	for {
+		ctx := context.Background()
+		var wg sync.WaitGroup
+		var matches []*pb.Match
+
+		// Fetch matches for each profile. Another variant to try could also be to
+		// set all profiles on a single FetchMatches call.
+		for _, p := range mprofiles {
+			wg.Add(1)
+			p := p
+			go func(wg *sync.WaitGroup) {
+				defer wg.Done()
+				results, err := doFetch(ctx, be, p)
+				if err != nil {
+					logger.Errorf("FetchMatches failed, got %v", err)
+					return
+				}
+
+				matches = append(matches, results...)
```

Comment:
I think the vanilla slice in Go is not concurrency safe

Reply:
Some patterns are safe concurrently (reads and editing indices known to not overlap), but append is not.  I recommend using channels.  This simplest one here is probably results := make(chan []*pb.Match, len(mprofiles), buffering the results so you don't need another go routine to pull them out.  Using this you can actually get rid the the waitgroup:

for i := 0; i < len(mprofiles); i++ {
  matches = append(matches, results...)
}

Reply:
actually, I was mixing up things in my head.  Buffered is good if you still have the wait group, no need to make it buffered if you replace the waitgroup.